core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel.hpp

```diff
@@ -142,8 +534,11 @@ struct ParallelReduceSpecialize<FunctorType, Kokkos::RangePolicy<PolicyArgs...>,
     const typename PolicyType::member_type begin = p.begin();
     const typename PolicyType::member_type end   = p.end();
 
+    if (end <= begin) return;
+
     ValueType result = ValueType();
-#pragma omp target teams distribute parallel for num_teams(512) map(to:f) map(tofrom:result) reduction(+: result)
+#pragma omp target teams distribute parallel for num_teams(512) \
```

Comment:
Why do you fix the number of teams to 512? Shouldn't this depend on the hardware.

Reply:
Yes it does. But I currently have no way of figuring this out properly. There are TONS of things which need fixing later ...